packages/rehype-shiki/src/plugin.mjs

```diff
@@ -53,143 +53,149 @@ function isCodeBlock(node) {
   );
 }
 
-export default function rehypeShikiji() {
-  return function (tree) {
-    visit(tree, 'element', (_, index, parent) => {
-      const languages = [];
-      const displayNames = [];
-      const codeTabsChildren = [];
-
-      let defaultTab = '0';
-      let currentIndex = index;
-
-      while (isCodeBlock(parent?.children[currentIndex])) {
-        const codeElement = parent?.children[currentIndex].children[0];
-
-        const displayName = getMetaParameter(
-          codeElement.data?.meta,
-          'displayName'
-        );
-
-        // We should get the language name from the class name
-        if (codeElement.properties.className?.length) {
-          const className = codeElement.properties.className.join(' ');
-          const matches = className.match(/language-(?<language>.*)/);
-
-          languages.push(matches?.groups.language ?? 'text');
-        }
-
-        // Map the display names of each variant for the CodeTab
-        displayNames.push(displayName?.replaceAll('|', '') ?? '');
-
-        codeTabsChildren.push(parent?.children[currentIndex]);
-
-        // If `active="true"` is provided in a CodeBox
-        // then the default selected entry of the CodeTabs will be the desired entry
-        const specificActive = getMetaParameter(
-          codeElement.data?.meta,
-          'active'
-        );
+function getLanguage(codeBlockNode) {
+  if (codeBlockNode?.children[0]?.properties?.className) {
+    return codeBlockNode.children[0].properties.className
+      ?.find(cn => cn.startsWith(languagePrefix))
+      ?.substring(languagePrefix.length);
+  }
+  return undefined;
+}
 
-        if (specificActive === 'true') {
-          defaultTab = String(codeTabsChildren.length - 1);
+export default function rehypeShikiji() {
+  return tree => {
+    if (!tree || !tree.children) {
+      return tree;
+    }
+
+    // First pass: Group adjacent code blocks into <CodeTabs>
+    const newChildren = [];
+    let i = 0;
+    while (i < tree.children.length) {
```

Comment:
I'm not a _huge_ fan that we are using a `while` loop and a `visit`, is there an alternative?